launcher/src/backend/Monitoring.js

```diff
@@ -36,6 +48,60 @@ export class Monitoring {
     await this.serviceManagerProm.nodeConnection.logout();
   }
 
+  // Jobs to handle on login
+  async login(remoteHost) {
+    await this.cleanup();
+    this.nodeConnection.nodeConnectionParams = remoteHost;
+    this.nodeConnectionProm.nodeConnectionParams = remoteHost;
+    await this.nodeConnection.establish();
+    await this.nodeConnectionProm.establish();
+    this.isLoggedIn = true;
+    await this.startGlobalMonitoringCacheBackgroundWorker();
+  }
+
+  // Jobs to handle on logout
+  async logout() {
+    await this.stopGlobalMonitoringCacheBackgroundWorker();
+    await this.cleanup();
+  }
+
+  // Refresh global monitoring cache on request
+  async refreshGlobalMonitoringCache() {
+    const uxtsNow = Math.floor(Date.now() / 1000);
+    const elapsedSeconds = uxtsNow - this.globalMonitoringCache.isRefreshing;
+    const refreshTimeoutInSeconds = 30; // allow further refresh at least after 30 seconds
+    const refreshTimeoutReached = elapsedSeconds > refreshTimeoutInSeconds;
+    if (!this.globalMonitoringCache.isRefreshing || refreshTimeoutReached) {
+      this.globalMonitoringCache.isRefreshing = uxtsNow;
+      if (!this.globalMonitoringCache.nodestatsInitialized) {
+        this.getNodeStats(); // dont cache but initialize nodestats in background once for faster page lodaing afterwards
+        this.globalMonitoringCache.nodestatsInitialized = true;
+      }
+      this.globalMonitoringCache.storagestatus = await this.getStorageStatus(true);
+      this.globalMonitoringCache.isRefreshing = 0;
+    }
+  }
+
+  // Start global monitoring cache refreshing periodically each X seconds async in background
+  // Therefore no "await" is used for implemented calls (otherwise the login is blocked)!
+  async startGlobalMonitoringCacheBackgroundWorker() {
+    this.refreshGlobalMonitoringCache();
+    this.globalMonitoringCache.intervalHandler = setInterval(
+      function (me) {
+        me.refreshGlobalMonitoringCache();
+      },
+      this.globalMonitoringCache.refreshIntervalSeconds * 1000,
+      this
```

Comment:
do you need to pass `this`?
I think it is possible to access just `this.refreshGlobalMonitoringCache()` instead of `me.refreshGlobalMonitoringCache()`

Reply:
yes, i had to add it because `this` was unavailable inside `setInterval`, resulting in the followign error:
![grafik](https://github.com/stereum-dev/ethereum-node/assets/107847185/167a1263-4ffb-44b0-9913-da11def076cc)